pkg/operation/botanist/network.go

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) 2018 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package botanist
+
+import (
+	"context"
+	"fmt"
+	"github.com/gardener/gardener/pkg/utils/kubernetes/health"
+	"github.com/gardener/gardener/pkg/utils/retry"
+	"time"
+
+	gardencorev1alpha1 "github.com/gardener/gardener/pkg/apis/core/v1alpha1"
+	gardencorev1alpha1helper "github.com/gardener/gardener/pkg/apis/core/v1alpha1/helper"
+	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
+	controllerutils "github.com/gardener/gardener/pkg/controllermanager/controller/utils"
+	"github.com/gardener/gardener/pkg/operation/common"
+	kutil "github.com/gardener/gardener/pkg/utils/kubernetes"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// NetworkDefaultTimeout is the default timeout and defines how long Gardener should wait
+// for a successful reconciliation of a network resource.
+const NetworkDefaultTimeout = 10 * time.Minute
+
+// DeployNetwork creates the `Network` extension resource in the shoot namespace in the seed
+// cluster. Gardener waits until an external controller did reconcile the cluster successfully.
+func (b *Botanist) DeployNetwork(ctx context.Context) error {
+	var (
+		lastOperation                = b.Shoot.Info.Status.LastOperation
+		creationPhase                = lastOperation != nil && lastOperation.Type == gardencorev1alpha1.LastOperationTypeCreate
+		requestNetworkReconciliation = creationPhase || controllerutils.HasTask(b.Shoot.Info.Annotations, common.ShootTaskDeployNetwork)
```

Comment:
This is not needed. I suggest "copying" the deploy function from the worker resource, for example.